app/vlinsert/journald/journald.go

```diff
@@ -273,30 +273,28 @@ func readJournaldLogEntry(streamName string, lr *insertutil.LineReader, lmp inse
 			fb.name = append(fb.name[:0], line...)
 			name = bytesutil.ToUnsafeString(fb.name)
 
-			fb.value = fb.value[:0]
-			for len(fb.value) < 8 {
-				if !lr.NextLine() {
-					if err := lr.Err(); err != nil {
-						return fmt.Errorf("cannot read value size: %w", err)
-					}
-					return fmt.Errorf("unexpected end of stream while reading value size")
+			if !lr.NextLine() {
+				if err := lr.Err(); err != nil {
+					return fmt.Errorf("cannot read value size: %w", err)
 				}
-				fb.value = append(fb.value, lr.Line...)
-				fb.value = append(fb.value, '\n')
+				return fmt.Errorf("unexpected end of stream while reading value size")
 			}
+			fb.value = append(fb.value[:0], lr.Line...)
+			fb.value = append(fb.value, '\n')
 			size := binary.LittleEndian.Uint64(fb.value[:8])
```

Comment:
This code can panic if `lr.NextLine()` returns a line shorter than 8 bytes. For example, if the encoded value length contains newline char (or multiple chars). The previous `for ...` loop was intended for handling this case.

Reply:
Ah that makes sense, completely forgot that that was a possibility